### DISCUSSION TITLE
question about sqlalchemy model

### DISCUSSION THREAD

**Discussion by JunkyardCat** [sqlalchemy/sqlalchemy#7397] [Usage Questions]
please see below my user model
the def 
```
models.py
Class User(Base):
    name = Column(String)
    key = Column(String)
    salt = Column(String)

    def check_password(self,input_password):
    <some operations>

```

next i have a code that queries all the users and check thier password

``` result = session.query(models.User).filter_by(name=val['username']).first()```

when i run the next statement it errors saying theres no such thing as check_password

```result.check_password()```

can anyone tell me why this is?

**Comment by CaselIT** [comment]
Hi,

Without more information, like a stack trace it's a bit hard to guess what'a happening
